src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressByMessageFilterTest.java

```diff
@@ -0,0 +1,165 @@
+        config.addProperty("messageFormat", "('[a-z]{1}')");
+
+        final String[] expectedViolations = {
+            "25:9: " + getCheckMessage(RequireThisCheck.class, MSG_METHOD, "foo", ""),
+        };
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter.java"),
+                expectedViolations);
+    }
+
+    @Test
+    public void testMatchMessageMethod() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(SuppressByMessageFilter.class);
+        config.addProperty("messageFormat", "('foo')");
+
+        final String[] expectedViolations = {
+            "17:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "a", ""),
+            "18:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "b", ""),
+            "23:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "c", ""),
+        };
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter.java"),
+                expectedViolations);
+    }
+
+    @Test
+    public void testNoMatchMessage() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(SuppressByMessageFilter.class);
+        config.addProperty("messageFormat", "Nonsense1234Regex");
+
+        final String[] expectedViolations = {
+            "17:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "a", ""),
+            "18:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "b", ""),
+            "19:9: " + getCheckMessage(RequireThisCheck.class, MSG_METHOD, "foo", ""),
+            "23:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "c", ""),
+        };
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter.java"),
+                expectedViolations);
+    }
+
+    @Test
+    public void testMatchCheckAndMessage() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(SuppressByMessageFilter.class);
+        config.addProperty("checkFormat", "RequireThisCheck");
+        config.addProperty("messageFormat", "needs");
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter.java"),
+                CommonUtil.EMPTY_STRING_ARRAY);
+    }
+
+    private void verifySuppressed(DefaultConfiguration config, String fileName,
+                                  String... expectedViolations) throws Exception {
+        final DefaultConfiguration requireThisConfig =
+                createModuleConfig(RequireThisCheck.class);
+        requireThisConfig.addProperty("validateOnlyOverlapping", "false");
+
+        final DefaultConfiguration treewalkerConfig = createModuleConfig(TreeWalker.class);
+        treewalkerConfig.addChild(requireThisConfig);
+        final DefaultConfiguration checkerConfig = createRootConfig(treewalkerConfig);
+        checkerConfig.addChild(config);
+
+        verify(checkerConfig, fileName, expectedViolations);
+        verifyFilterWithInlineConfigParser(fileName, expectedViolations);
```

Comment:
Why are you using 2 verify methods in combination? `WithInlineConfigParser` should be the primary verify over anything else.

Reply:
Okay. Changes Done.